src/sparse/linalg/ordering.rs

```diff
@@ -13,74 +11,512 @@ pub struct Ordering<I> {
     pub connected_parts: Vec<usize>,
 }
 
-pub fn cuthill_mckee<N, I, Iptr>(mat: CsMatViewI<N, I, Iptr>) -> Ordering<I>
+pub mod start {
+    use indexing::SpIndex;
+    use sparse::CsMatViewI;
+
+    /// This trait abstracts over possible strategies to choose a starting vertex for the Cutihll-McKee algorithm.
+    /// Common strategies are provided.
+    pub trait Strategy<N, I, Iptr>
+    where
+        N: PartialEq,
+        I: SpIndex,
+        Iptr: SpIndex,
+    {
+        /// Contract: This function must always be called with at least one unvisited vertex left.
+        fn find_start_vertex(
+            &mut self,
+            visited: &[bool],
+            degrees: &[usize],
+            mat: &CsMatViewI<N, I, Iptr>,
+        ) -> usize;
+    }
+
+    /// This strategy chooses predetermined starting vertices.
```

Comment:
Is this predetermined strategy something common? As a first thought I find it hard to find a sequence of vertices that should work. Or maybe that's a good strategy when one knows very well the sparse structure? In that case, maybe a test case showing a relevant example would be good. Not a priority if that's a lot of work though.

Reply:
Acctually, it probably is not. It is definitely something which *could* be of interest, only to a small subset of problems though, when somebody knows the structure of the matrix very well. For that case we would then, however, have to enable choosing multiple starting vertices as well.
That does seem like to much. I will remove this strategy, pointing out in the trait documentation custom strategies can be implemented by the user, with a note that we would love to be informed about what custom strategies users implement.